x/oracle/keeper/update_exchange_rates.go

```diff
@@ -27,21 +27,43 @@ func (k Keeper) UpdateExchangeRates(ctx sdk.Context) {
 	params, _ := k.Params.Get(ctx)
 	k.clearVotesAndPreVotes(ctx, params.VotePeriod)
 	k.updateWhitelist(ctx, params.Whitelist, whitelistedPairs)
+	k.registerAbstainsByOmission(ctx, len(params.Whitelist), validatorPerformances)
+	return validatorPerformances
 }
 
-// registerMissedVotes it parses all validators performance and increases the missed vote of those that did not vote.
-func (k Keeper) registerMissedVotes(ctx sdk.Context, whitelistedPairs set.Set[asset.Pair], validatorPerformances types.ValidatorPerformances) {
+// registerMissedVotes it parses all validators performance and increases the
+// missed vote of those that did not vote.
+func (k Keeper) registerMissedVotes(
+	ctx sdk.Context,
+	whitelistedPairs set.Set[asset.Pair],
+	validatorPerformances types.ValidatorPerformances,
+) {
 	for _, validatorPerformance := range validatorPerformances {
-		if int(validatorPerformance.WinCount) == len(whitelistedPairs) {
-			continue
+		if int(validatorPerformance.MissCount) > 0 {
+			k.MissCounters.Insert(
+				ctx, validatorPerformance.ValAddress,
+				k.MissCounters.GetOr(ctx, validatorPerformance.ValAddress, 0)+1)
+			k.Logger(ctx).Info("vote miss", "validator", validatorPerformance.ValAddress.String())
 		}
+	}
+}
 
-		k.MissCounters.Insert(ctx, validatorPerformance.ValAddress, k.MissCounters.GetOr(ctx, validatorPerformance.ValAddress, 0)+1)
-		k.Logger(ctx).Info("vote miss", "validator", validatorPerformance.ValAddress.String())
+func (k Keeper) registerAbstainsByOmission(
```

Comment:
I don't think we need this function. The return type of `UpdateExchangeRates` isn't used anywhere so we don't need to do anything after rewarding validators and updating miss counters.